File: Python/Labs/Module 1/Lab 2/Q2 - Queue Using Two Stacks/main.py
s1=[]
s2=[]

def enqueue(item):
    s1.append(item)

def dequeue():
    if not s2:
        if not s1:
            return None
        while s1:
            s2.append(s1.pop())
    return s2.pop(0) if s2 else None

def peek():
    if not s2:
        if not s1:
            return None
        while s1:
            s2.append(s1.pop())
    return s2[0] if s2 else None

inp=input().split(",")
a=[]
for i in inp:
    x=i.split()
    a.append(x)
for i in a:
    if i[0]=='1':
        enqueue(i[1])
    elif i[0]=='2':
        dequeue()
    else:
        print(peek())
                                                                                                                            

File: Java/Labs/Module 1/Lab 2/Q2 - Queue Using Two Stacks/Main.java
import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        String i=s.nextLine();
        p(i.split(","));
        s.close();
    }
    public static void p(String[] q){
        Stack<Integer> s1=new Stack<>();
        Stack<Integer> s2=new Stack<>();
        for(String qy:q){
            String[] p=qy.trim().split(" ");
            int t=Integer.parseInt(p[0]);
            if(t==1){
                int v=Integer.parseInt(p[1]);
                s1.push(v);
            } else if(t==2){
                if(s2.isEmpty()){
                    while(!s1.isEmpty()){
                        s2.push(s1.pop());
                    }
                }
                if(!s2.isEmpty()){
                    s2.pop();
                }
            } else if(t==3){
                if(s2.isEmpty()){
                    while(!s1.isEmpty()){
                        s2.push(s1.pop());
                    }
                }
                if(!s2.isEmpty()){
                    System.out.println(s2.peek());
                }
            }
        }
    }
}
                                                                                                                            